src/alt.js

```diff
@@ -460,7 +460,7 @@ class Alt {
   takeSnapshot(...storeNames) {
     const state = snapshot(this, ...storeNames)
     if (this[LAST_SNAPSHOT]) {
-      assign(this[LAST_SNAPSHOT], state)
+      this[LAST_SNAPSHOT] = JSON.stringify(assign(JSON.parse(this[LAST_SNAPSHOT]), JSON.parse(state)))
```

Comment:
Okay, so that did not clean up as nicely as I was hoping.  My first attempt I ended up with this:

```js
  takeSnapshot(...storeNames) {
    const state = snapshot(this, ...storeNames)
    if (this[LAST_SNAPSHOT]) {
      this[LAST_SNAPSHOT] = JSON.stringify(assign(JSON.parse(this[LAST_SNAPSHOT]), state))
    } else {
      this[LAST_SNAPSHOT] = JSON.stringify(state)
    }
    return JSON.stringify(state)
  }
```

which I refactored into:

```js
  takeSnapshot(...storeNames) {
    const state = snapshot(this, ...storeNames)
    this[LAST_SNAPSHOT] = JSON.stringify(
      assign(JSON.parse(this[LAST_SNAPSHOT] || '{}'), state)
    )
    return JSON.stringify(state)
  }
```